Create a BaseGeometry instance
=======


::

    >>> BaseGeometry = __import__('7-base_geometry').BaseGeometry
    >>> bg = BaseGeometry()
    >>> bg.integer_validator('casa', 2)
    

Non-Numbers
===========

empty arguments
::

    >>> bg.integer_validator()
    Traceback (most recent call last):
    	        ...
    TypeError: integer_validator() takes exactly 3 arguments (1 given)


string as arguments
::

    >>> bg.integer_validator('name', 'John')
    Traceback (most recent call last):
    	        ...
    TypeError: name must be an integer

None as arguments
::

    >>> bg.integer_validator('age', 0)
    Traceback (most recent call last):
    	        ...
    ValueError: age must be greater than 0

args

::
    >>> bg.integer_validator('distance', -4)
    Traceback (most recent call last):
    	        ...
    ValueError: distance must be greater than 0



    >>> bg.integer_validator("float", 3.5)
    Traceback (most recent call last):
	...
    TypeError: float must be an integer

    >>> bg.integer_validator("infinity", float('inf'))
    Traceback (most recent call last):
    ...
    TypeError: infinity must be an integer


    >>> bg.integer_validator("nan", float('nan'))
    Traceback (most recent call last):
    ...
    TypeError: nan must be an integer


    >>> bg.integer_validator("list", [1, 2, 3])
    Traceback (most recent call last):
    ...
    TypeError: list must be an integer


    >>> bg.integer_validator("tuple", (1, 2, 3))
    Traceback (most recent call last):
    ...
    TypeError: tuple must be an integer


    >>> bg.integer_validator("bool", True)
    Traceback (most recent call last):
    ...
    TypeError: bool must be an integer


    >>> bg.integer_validator("set", {1, 2})
    Traceback (most recent call last):
    ...
    TypeError: set must be an integer


    >>> bg.integer_validator("none", None)
    Traceback (most recent call last):
    ...
    TypeError: none must be an integer


    >>> bg.integer_validator()
    Traceback (most recent call last):
    ...
    TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'


    >>> bg.integer_validator("arg")
    Traceback (most recent call last):
    ...
    TypeError: integer_validator() missing 1 required positional argument: 'value'
